test/v3/USTBv3TokenStorageLayoutTests.t.sol

```diff
@@ -0,0 +1,151 @@
+ *  Slot 758-854: SuperstateToken.__additionalFieldsGap
+ */
+contract USTBv3TokenStorageLayoutTests is SuperstateTokenStorageLayoutTestBase {
+    function initializeExpectedTokenVersions() public override {
+        oldTokenVersion = "2";
+        newTokenVersion = "3";
+    }
+
+    function initializeOldToken() public override {
+        USTBv1 oldTokenImplementation = new USTBv1(address(this), perms);
+        tokenProxy = new TransparentUpgradeableProxy(address(oldTokenImplementation), address(this), "");
+        tokenProxyAdmin = ProxyAdmin(getAdminAddress(address(tokenProxy)));
+
+        // wrap in ABI to support easier calls
+        oldToken = USTBv1(address(tokenProxy));
+
+        oldToken.initialize("Superstate Short Duration US Government Securities Fund", "USTB");
+
+        USTBv2 oldTokenImplementationV2 = new USTBv2(address(this), perms);
+        tokenProxyAdmin.upgradeAndCall(
+            ITransparentUpgradeableProxy(address(tokenProxy)), address(oldTokenImplementationV2), ""
+        );
+
+        // wrap in ABI to support easier calls
+        oldToken = USTBv2(address(tokenProxy));
+
+        ISuperstateTokenV2(address(oldToken)).initializeV2();
+
+        currentToken = oldToken;
+    }
+
+    function upgradeAndInitializeNewToken() public override {
+        // Now upgrade to V3
+        USTB newTokenImplementation = new USTB(perms);
+        tokenProxyAdmin.upgradeAndCall(
+            ITransparentUpgradeableProxy(address(tokenProxy)), address(newTokenImplementation), ""
+        );
+
+        /*
+            At this point, owner() is 0x00 because the upgraded contract has not
+            initialized.
+
+            admin() is the same from the prior version of the contract
+        */
+
+        // v3 requires no new initialize function
+        newToken = USTB(address(tokenProxy));
+
+        currentToken = newToken;
+    }
+
+    function assertSuperstateTokenStorageLayout(bool hasUpgraded) public override {
+        super.assertSuperstateTokenStorageLayout(hasUpgraded);
+
+        if (hasUpgraded) {
+            bytes32 maximumOracleDelaySlot = keccak256(abi.encode(eve, uint256(755)));
+            uint256 maximumOracleDelaySlotValue = uint256(vm.load(address(tokenProxy), maximumOracleDelaySlot));
+            uint256 expectedMaximumOracleDelayValue = 0;
+            assertEq(maximumOracleDelaySlotValue, expectedMaximumOracleDelayValue);
+
+            bytes32 superstateOracleSlot = keccak256(abi.encode(eve, uint256(756)));
+            uint256 superstateOracleSlotValue = uint256(vm.load(address(tokenProxy), superstateOracleSlot));
+            uint256 expectedSuperstateOracleValue = 0;
+            assertEq(superstateOracleSlotValue, expectedSuperstateOracleValue);
+
+            bytes32 supportedStablecoinsSlot = keccak256(abi.encode(eve, uint256(757)));
+            uint256 supportedStablecoinsSlotValue = uint256(vm.load(address(tokenProxy), supportedStablecoinsSlot));
```

Comment:
might be more valuable to actually mutate state so the values aren't 0. Right now, the test isn't really asserting anything because the values would be 0 anyways due to the `_additonalFieldsGap`

Reply:
I'll be mutating state and testing in tests i write in a follow up PR